Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -31,115 +31,247 @@ using namespace Windows::Storage;
 using namespace Windows::Graphics::Imaging;
 using namespace Windows::Media::Core;
 
-// This is global because we need to hold the reference to sourcegroups. The sourceinfos are holding weak references to this.
-IVectorView<MediaFrameSourceGroup> sourceGroups;  
-
-//
-// Iterates through source groups and filter-out the frame sources which have IR, depth and other sources which we cannot consume in this app
-//
-IVector<MediaFrameSourceInfo> GetFilteredSourceGroupList()
+class CameraHelper
 {
-    auto filteredSourceInfos = single_threaded_vector<MediaFrameSourceInfo>();
-    sourceGroups = MediaFrameSourceGroup::FindAllAsync().get();
-    auto sourceGroupIter = sourceGroups.First();
-    while (sourceGroupIter.HasCurrent())
+
+    // We need to hold the reference to sourcegroups. The sourceinfos are holding weak references to this.
+    IVectorView<MediaFrameSourceGroup> m_SourceGroups;
+
+    // This can be used as a mock preview
+    MediaFrameReader m_PreviewFrameReader;
+
+    //
+    // Iterates through source groups and filter-out the frame sources which have IR, depth and other sources which we cannot consume in this app
+    //
+    IVector<MediaFrameSourceInfo> GetFilteredSourceGroupList()
     {
-        auto sourceInfos = sourceGroupIter.Current().SourceInfos();
-        auto sourceInfoIter = sourceInfos.First();
-        // Iterate through sources and filter-out the IR, depth and other sources which we cannot consume in this app
-        while (sourceInfoIter.HasCurrent())
+        auto filteredSourceInfos = single_threaded_vector<MediaFrameSourceInfo>();
+        m_SourceGroups = MediaFrameSourceGroup::FindAllAsync().get();
+        auto sourceGroupIter = m_SourceGroups.First();

```

Comment:
catch case where there are no source group returned

Reply:
In this case, this function simply returns a empty vector. Further , this case is handled at line 215. 

---
In reply to: [286775667](https://github.com/microsoft/Windows-Camera/pull/10#discussion_r286775667) [](ancestors = 286775667)